chainstate/src/detail/median_time.rs

```diff
@@ -185,11 +186,11 @@ mod test {
             .unwrap();
 
             // we use unordered block times, and ensure that the median will be in the right spot
-            let block1_time = current_time.load(Ordering::SeqCst) + 1;
-            let block2_time = current_time.load(Ordering::SeqCst) + 10;
-            let block3_time = current_time.load(Ordering::SeqCst) + 17;
-            let block4_time = current_time.load(Ordering::SeqCst) + 18;
-            let block5_time = current_time.load(Ordering::SeqCst) + 20;
+            let block1_time = current_time.load() + 1;
+            let block2_time = current_time.load() + 10;
+            let block3_time = current_time.load() + 17;
+            let block4_time = current_time.load() + 18;
+            let block5_time = current_time.load() + 20;
```

Comment:
There's something something simple I wanted to fix these but never had the time. Would be great if you could. There's a story there.

When I developed the BlockIndexHistoryIterator, this test was meant to have non-sequential time. But then when PoS was developed, we added a restriction to check_block to enforce time-ordering of blocks, which then was reverted and this time-ordering was enforced in PoS consensus validity check. So this current ordering doesn't cover the case where consecutive blocks have unordered timestamps.

Someone should revert the order of these times and ensure that the old block times are used, where no order is enforced. Please check the history to see when this change was made a couple of months ago.

Reply:
Done in PR #998 (if I've understood you correctly)

Reply:
Yep. Thanks.